src/Strategy/Erlang/ConfigParser.hs

```diff
@@ -0,0 +1,51 @@
+module Strategy.Erlang.ConfigParser (
+  parseConfig,
+  parseErlValue,
+) where
+
+import Data.Functor (void)
+import Data.Text (Text)
+import Data.Void
+import Text.Megaparsec
+import Text.Megaparsec.Char
+import qualified Text.Megaparsec.Char.Lexer as L
+
+type Parser = Parsec Void Text
+
+data ErlValue
+  = Atom Text
+  | Array [ErlValue]
+  | Pair Text ErlValue
+
+parseConfig :: Parser [ErlValue]
+parseConfig = do
+  scn
+  many (parsePair <* char '.')
+
+parseErlValue :: Parser ErlValue
+parseErlValue = parseAtom <|> parseArray <|> parsePair
+
+parseAtom :: Parser ErlValue
+parseAtom = Atom <$> lexeme (takeWhile1P (Just "atom") (== ','))
+
+parseArray :: Parser ErlValue
+parseArray = lexeme (Array <$> between (char '[') (char ']') (many parseErlValue))
+
+parsePair :: Parser ErlValue
+parsePair = do
+  cc '{'
+  (Atom atom) <- parseAtom
+  cc ','
+  value <- parseErlValue
+  cc '}'
+  pure $ Pair atom value
+
+-- Char consumer (also consumes space)
+cc :: Char -> Parser ()
+cc = void <$> lexeme . char
```

Comment:
what's the monomorphized type of `<$>` here? the answer may surprise you.

in ghci:
```hs
>>> :t void `_something` lexeme . char :: Char -> Parser ()
```

is there anything more familiar that has that type?

Reply:
That was the most clickbait PR comment I've ever received.

Reply:
Also, if you're specifically pointing me somewhere, I'm not seeing it.  I'm sure there's more than one way to do this, but I can't think of any way which is more readable, convenient, or general.  What am I missing here?